edx_proctoring/handlers.py

```diff
@@ -127,15 +135,16 @@ def on_attempt_changed(sender, instance, signal, **kwargs):
     else:
         # remove the attempt on the backend
         # timed exams have no backend
-        backend = get_backend_provider(name=instance.proctored_exam.backend)
-        if backend:
-            result = backend.remove_exam_attempt(instance.proctored_exam.external_id, instance.external_id)
-            if not result:
-                log.error(
-                    'Failed to remove attempt_id=%s from backend=%s',
-                    instance.id,
-                    instance.proctored_exam.backend,
-                )
+        if instance.proctored_exam.is_proctored:
```

Comment:
Only remove from the backend if it is a proctored exam, so this does fix a bug where instructors wouldn't be able to delete a timed exam.

[**Question**] For the case of PSI, do we care if removing attempts blows up?